language/stdlib/modules/Diem.move

```diff
@@ -606,14 +620,27 @@ module Diem {
         aborts_if info.total_value < to_burn with Errors::LIMIT_EXCEEDED;
         aborts_if info.preburn_value < to_burn with Errors::LIMIT_EXCEEDED;
     }
-
     spec schema BurnWithResourceCapEnsures<CoinType> {
         preburn: Preburn<CoinType>;
         ensures spec_currency_info<CoinType>().total_value
                 == old(spec_currency_info<CoinType>().total_value) - old(preburn.to_burn.value);
         ensures spec_currency_info<CoinType>().preburn_value
                 == old(spec_currency_info<CoinType>().preburn_value) - old(preburn.to_burn.value);
     }
+    spec schema BurnWithResourceCapEmits<CoinType> {
+        preburn: Preburn<CoinType>;
+        preburn_address: address;
+        let info = spec_currency_info<CoinType>();
+        let currency_code = spec_currency_code<CoinType>();
+        let handle = info.burn_events;
+        // TODO: `old(..)` is not supported in the pack expression.
+        // let msg = BurnEvent {
+        //     amount: old(preburn.to_burn.value),
+        //     currency_code,
+        //     preburn_address,
+        // };
+        // emits msg to handle if !info.is_synthetic;
+    }
```

Comment:
If you haven't filed an issue about this limitation, please do so.

Reply:
This is not a bug but by design. You can file an issue to change the design, but it won't be easy.

Reply:
Wolfgang told me that `old(..)` actually works in a pack expression, but not in `let`. So, I finished these TODO specs avoiding using `let`. The issue of `old(..)` with `let` has been reported here: https://github.com/diem/diem/issues/5739.